docs/observability.asciidoc

```diff
@@ -0,0 +1,251 @@
+[[observability]]
+== Observability
+
+The client does not provide a default logger, but instead it offers an event emitter interfaces to hook into internal events, such as `request` and `response`.
+
+Correlating those events can be quite hard, especially if your applications have a large codebase with many events happening at the same time.
+
+To help you with this, the client offers you a correlation id system and other features, let's see them in action.
+
+=== Events
+The client is an event emitter, this means that you can listen for its event and add additional logic to your code, without need to change the client internals or your normal usage. +
+You can find the events names by access the `events` key of the client.
+
+[source,js]
+----
+const { events } = require('@elastic/elasticsearch')
+console.log(events)
+----
+
+The event emitter functionality can be useful if you want to log every request, response and error that is happening during the use of the client.
+
+[source,js]
+----
+const logger = require('my-logger')()
+const { Client } = require('@elastic/elasticsearch')
+const client = new Client({ node: 'http://localhost:9200' })
+
+client.on('response', (err, req) => {
+  if (err) {
+    logger.error(err)
+  } else {
+    logger.info(req)
+  }
+})
+----
+
+The client emits the following events:
+[cols=2*]
+|===
+|`request`
+a|Emitted before to send the actual request to Elasticsearch.
+[source,js]
+----
+client.on('request', (err, req) => {
+  console.log(err, req)
+})
+----
+
+|`response`
+a|Emitted before to send the actual request to Elasticsearch.
+[source,js]
+----
+client.on('response', (err, req) => {
+  console.log(err, req)
+})
+----
+
+|`sniff`
+a|Emitted when the clients ends a sniffing request.
+[source,js]
+----
+client.on('sniff', (err, req) => {
+  console.log(err, req)
+})
+----
+
+|`resurrect`
+a|Emitted if the client is able to resurrect a dead node.
+[source,js]
+----
+client.on('resurrect', (err, req) => {
+  console.log(err, req)
+})
+----
+
+|===
+
+The values of `req` in `request`, `response` and `sniff` will be:
+[source,ts]
+----
+  body: any;
+  statusCode: number | null;
+  headers: anyObject | null;
+  warnings: string[] | null;
+  meta: {
+    context: any;
+    name: string;
+    request: {
```

Comment:
Agree that `req` might not be a good name, I'll use `result`, since is the same used by the API methods.

> what about calling these requestParams, requestOptions, and requestId?

I prefer to have a nested object, since it is more future proof and I can easily extend it in the future.